### ISSUE TITLE
Memory Error When Writing ToBinary With "AllProps"

### ISSUE THREAD

**Issue by atom-moyer** [rdkit/rdkit#2352]
Hi,

I am a daily python developer using the current version of RDKit, and today, I have run into an edge-case bug.

My workflow involves me iteratively saving the state of RDKit molecules into a serialized format (atom properties, molecule properties, and conformations are important).

I have found, if I run the following lines of code, I recieve a memory error:

```
import rdkit                                                                    
from rdkit.Chem.rdmolfiles import MolFromSmiles         
from rdkit.Chem.AllChem import EmbedMultipleConfs       
                        
mol = MolFromSmiles('CCCCCC')                                                   
                        
EmbedMultipleConfs(mol)                                                         
-> <rdkit.rdBase._vecti at 0x7ffff437f630>

rdkit.Chem.SetDefaultPickleProperties(rdkit.Chem.PropertyPickleOptions.AllProps)

mol2 = rdkit.Chem.Mol(mol.ToBinary())                                           

EmbedMultipleConfs(mol2)                                                        
-> <rdkit.rdBase._vecti at 0x7ffff437f2d0>

mol2.ToBinary()                                                                 
----------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
<ipython-input-9-f78d1b445ca1> in <module>
----> 1 mol2.ToBinary()

MemoryError: 
```

Is there a better way to serialize molecules including all of their property data which is more supported, or did I just run into a weird edge case?

Note: This happens if I run with clearConfs = (True or False).

Adam

**Comment by atom-moyer** [comment]
My work around is as follows for anyone else that comes across this:

```
mol3 = rdkit.Chem.Mol(mol.ToBinary())

for conformer in mol2.GetConformers():
    mol3.AddConformer(conformer) 

mol3.ToBinary()
```

**Comment by greglandrum** [comment]
Hi, which version of the RDKit are you using, with which version of python on which platform?

**Comment by iiitmjay** [comment]
As far as my knowledge is memory error occurs when there is not enough contiguous space, but I couldn't get what the issue here id

**Comment by atom-moyer** [comment]
Python 3.6.7
rdkit 2018.09.1 - via conda
Ubuntu 18.04.1 LTS

**Comment by greglandrum** [comment]
Thanks.
I am able to reproduce this with v2018.09.2 and python3.6 or python3.7.
Curiously, I am not able to reproduce it on master.
I'll look into it.

**Comment by greglandrum** [comment]
Here's a smaller reproducible:
```
import rdkit
from rdkit.Chem.rdmolfiles import MolFromSmiles
from rdkit.Chem.AllChem import EmbedMultipleConfs
mol = MolFromSmiles('CCCCCC')
rdkit.Chem.SetDefaultPickleProperties(
    rdkit.Chem.PropertyPickleOptions.AllProps)
mol2 = rdkit.Chem.Mol(mol.ToBinary())
EmbedMultipleConfs(mol2)
mol2.ClearComputedProps()
```

**Comment by greglandrum** [comment]
Ok, this was fixed on master as part of #2123 
Parts of that PR need to be backported to the 2018.09 release branch.

**Comment by atom-moyer** [comment]
Great!

There is a work around for now, so there is no rush.

**Comment by greglandrum** [comment]
Fixed in 8ee1899 and 5afbdb6 on the release branch.
